service/src/main/kotlin/io/provenance/explorer/service/AssetService.kt

```diff
@@ -226,13 +233,83 @@ class AssetService(
                     val data = markerClient.getMarkerDetail(marker)
                     MarkerCacheRecord.findByDenom(marker)?.apply {
                         if (data != null) this.status = data.status.toString()
-                        this.supply = accountService.getCurrentSupply(marker).toBigDecimal()
+                        this.supply = getCurrentSupply(marker).toBigDecimal()
                         this.lastTx = txTime.toDateTime()
                         this.data = data
                     }
                 }
             }
         }
+
+    fun getTotalAum() = runBlocking {
+        val baseMap = MarkerCacheRecord.find {
+            (MarkerCacheTable.status eq MarkerStatus.MARKER_STATUS_ACTIVE.name) and
+                (MarkerCacheTable.supply greater BigDecimal.ZERO)
+        }.associate { it.denom to it.supply }
+        val pricing = baseMap.keys.toList().chunked(50) { getPricingInfo(it) }.flatMap { it.toList() }.toMap()
+            .toMutableMap().also { it.putAll(getPricingForNhash()) }
+        baseMap.map { (k, v) -> (pricing[k] ?: BigDecimal.ZERO).multiply(v.toHashSupply(k)) }.sumOf { it }
+    }
+
+    fun getAumForList(denoms: Map<String, String>): BigDecimal {
+        val pricing =
+            denoms.keys.toList().chunked(50) { getPricingInfo(it) }.flatMap { it.toList() }.toMap().toMutableMap()
+                .also { it.putAll(getPricingForNhash()) }
+        return denoms
+            .map { (k, v) -> (pricing[k] ?: BigDecimal.ZERO).multiply(v.toBigDecimal().toHashSupply(k)) }
+            .sumOf { it }
+    }
+
+    fun getPricingInfoIn(denoms: List<String>) =
+        getPricingInfo(denoms).also { it.putAll(getPricingForNhash()) }
+
+    fun getPricingInfo(denoms: List<String>): MutableMap<String, BigDecimal> {
+        fun JSONObject.getDecimalOrNull(key: String) = try {
+            this.getBigDecimal(key)
+        } catch (e: Exception) {
+            BigDecimal.ZERO
+        }
+
+        val res = khttp.get(
+            url = "${props.pricingUrl}/api/v1/pricing/marker/denom/list",
+            params = mapOf("denom[]" to denoms.joinToString(","))
+        )
+
+        return if (res.statusCode == 200) {
+            try {
+                res.jsonArray.mapNotNull { ele ->
+                    if (ele is JSONObject) {
+                        ele.getString("markerDenom") to ele.getDecimalOrNull("usdPrice")
+                    } else null
+                }.toMap().toMutableMap()
+            } catch (e: Exception) {
+                mutableMapOf<String, BigDecimal>().also { logger.error("Error: $e") }
+            }
+        } else mutableMapOf<String, BigDecimal>().also { logger.error("Error reaching Pricing Engine: ${res.jsonObject}") }
+    }
+
+    fun getPricingForNhash(): Map<String, BigDecimal> {
+        val url =
+            "https://www.dlob.io/aggregator/external/api/v1/order-books/pb18vd8fpwxzck93qlwghaj6arh4p7c5n894vnu5g/daily-price"
```

Comment:
😃 I can check the price without having to log in!

Reply:
lol pretty much